hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1788,12 +1999,16 @@ int SaraNcpClient::enterDataMode() {
 
     CHECK(waitAtResponse(dataParser_, 5000));
 
-    if (ncpId() != PLATFORM_NCP_SARA_R410) {
+    if (ncpId() != PLATFORM_NCP_SARA_R410 && ncpId() != PLATFORM_NCP_SARA_R510) {
         // Ignore response
         // SARA R4 does not support this, see above
         CHECK(dataParser_.execCommand(20000, "ATH"));
     }
 
+    if (ncpId() == PLATFORM_NCP_SARA_R510) {
+        parser_.execCommand(10000, "AT+UPORTFWD=100,%d", UBLOX_DEFAULT_CID); // Enable full cone NAT support
```

Comment:
Yes, let's send this over second channel, same as everything in this function.

Reply:
I suppose if it replies with "OK" it'll probably work... but we'll need to repeat the NAT timeout testing.

Reply:
And yeah, we should actually check that it returns `OK` to us, otherwise we'll silently continue the connection process and we'll run into NAT timeout issues.

Reply:
The reason it doesn't check was because 3.11 errors on this command.. but I guess we should change it now to check :D

Reply:
Wrapped with CHECK_PARSER_OK()